server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -192,6 +198,61 @@ public void clear(String tenantId) {
         }
     }
 
+    @Override
+    public Trace getTraceFragment(String tenantId, String id) {
+        Span span = getSpan(tenantId, id);
+
+        Trace trace = spanToTrace(span);
+        trace.setNodes(recursiveTraceFragment(tenantId, id));
+
+        return trace;
+    }
+
+    private List<Node> recursiveTraceFragment(String tenantId, String spanId) {
+        List<Span> spanChildren = getChildren(tenantId, spanId);
+        if (spanChildren == null) {
+            return Collections.emptyList();
+        }
+
+        List<Node> nodes = new ArrayList<>();
+
+        for (Span span: spanChildren) {
+            if (span.serverSpan()) {
+                continue;
+            }
+
+            // TODO set uri
+            Consumer consumer = new Consumer(null, "http");
+            consumer.addInteractionId(span.getId());
+            consumer.setNodes(recursiveTraceFragment(tenantId, span.getId()));
+
+            nodes.add(consumer);
+        }
+
+        return nodes;
+    }
+
+    private Trace spanToTrace(Span span) {
+        if (span == null) {
+            throw new NullPointerException();
+        }
+
+        List<Property> spanProperties = span.properties();
+
+        Trace trace = new Trace();
+        trace.setId(span.getId());
+        trace.setStartTime(span.getTimestamp());
+        trace.setHostAddress(Span.ipv4Address(spanProperties));
+        trace.setProperties(new HashSet<>(span.properties()));
+
+//        trace.setBusinessTransaction()
+//        trace.setHostName()
+//        trace.setPrincipal()
+//        trace.setNodes()
+
+        return trace;
```

Comment:
This should also result in the top level node being created - if the span is a serverSpan() then it should be represented by a Consumer, a clientSpan, then a Producer, and anytihng else would be a Component. So probably best to have this logic in one place.